gyunuk/composestudy/app/src/main/java/com/bendeng/composestudy/ui/screen/home/HomeScreen.kt

```diff
@@ -0,0 +1,205 @@
+    ) {
+        Column(modifier = modifier) {
+            HomeSearchBar(
+                text = state.searchText,
+                onSearch = viewModel::searchPhotoList,
+                onUpdateSearchText = viewModel::updateSearchText,
+                onClearSearchText = viewModel::clearSearchText
+            )
+            ArticleList(images = state.photos)
+        }
+        LoadingDialog(show = state.loading)
+    }
+}
+
+@Composable
+fun ArticleList(
+    modifier: Modifier = Modifier,
+    images: List<DocumentData>,
+) {
+    LazyColumn(modifier = modifier.padding(vertical = 20.dp).fillMaxWidth()) {
+        items(items = images) { item ->
+            ArticleImage(item, modifier = Modifier.padding(top = 20.dp))
+        }
+    }
+}
+
+@Composable
+fun ArticleImage(
+    image: DocumentData,
+    modifier: Modifier = Modifier,
+) {
+    Surface(
+        modifier = modifier.fillMaxWidth().heightIn(120.dp),
+    ) {
+        Row(
+            verticalAlignment = Alignment.Top,
+        ) {
+            Box(modifier = Modifier.wrapContentSize()) {
+                RoundedGlideImage(
+                    imageModel = image.imageUrl,
+                    contentDescription = null,
+                    size = 120.dp
+                )
+            }
+
+            Spacer(modifier = Modifier.width(16.dp))
+            Column(
+                modifier = Modifier.fillMaxHeight().weight(1f),
+                verticalArrangement = Arrangement.spacedBy(20.dp)
+            ) {
+                Row(
+                    modifier = Modifier.heightIn(20.dp),
+                    verticalAlignment = Alignment.CenterVertically
+                ) {
+                    Image(
+                        painter = painterResource(R.drawable.ic_image),
+                        contentDescription = null,
+                        modifier = Modifier.size(20.dp)
+                    )
+                    Text(
+                        text = image.displaySitename,
+                        fontSize = 16.sp,
+                        modifier = Modifier.padding(horizontal = 8.dp),
+                        fontWeight = FontWeight.SemiBold
+                    )
+                }
+
+                Text(
+                    text = image.docUrl,
+                    fontSize = 10.sp,
+                    color = Color.Gray,
+                )
+
+                Text(
+                    text = DateTimeUtils.formatDateTime(image.datetime),
+                    fontSize = 10.sp,
+                    color = Color.Gray
+                )
+            }
+            Text(
+                text = image.collection,
+                color = Color.Blue,
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun HomeSearchBar(
+    modifier: Modifier = Modifier,
+    text: String,
+    onSearch: () -> Unit,
+    onUpdateSearchText: (String) -> Unit,
+    onClearSearchText: () -> Unit,
```

Comment:
```suggestion
    text: String,
    modifier: Modifier = Modifier,
    onSearch: () -> Unit,
    onUpdateSearchText: (String) -> Unit,
    onClearSearchText: () -> Unit,
```
저도 잘 못 지키는 컨벤션인데.. 매개변수로 필수 매개변수 다음으로 선택 매개변수 오는 것이 권장하는 방식입니다!
https://kotlinlang.org/docs/functions.html#default-arguments

Reply:
늘 컨벤션 지키는건 어렵네요.. 
필수 -> 비필수 파라미터 메모..